File: HireHubDemo/hirehub-referral/Models/JobApplication.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireHub.Referral.Models
{
    [Table("applications")]
    public class JobApplication
    {
        [Key]
        public long Id { get; set; }

        [Column("job_id")]
        public long JobId { get; set; }

        [Column("candidate_id")]
        public long CandidateId { get; set; }

        [Column("status")]
        public string Status { get; set; }

        [Column("applied_at")]
        public DateTime AppliedAt { get; set; }

        // Navigation properties
        public Candidate Candidate { get; set; }
        public Job Job { get; set; }
    }
}

File: HireHubDemo/HireHubDemo/hirehub-core/src/main/java/com/hirehub/core/entities/JobApplication.java
package com.hirehub.core.entities;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "applications")
@Data
public class JobApplication {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "job_id", nullable = false)
    private Job job;

    @ManyToOne
    @JoinColumn(name = "candidate_id", nullable = false)
    private Candidate candidate;

    @Enumerated(EnumType.STRING)
    private Status status = Status.APPLIED;

    @CreationTimestamp
    @Column(name = "applied_at", updatable = false)
    private LocalDateTime appliedAt;

    public enum Status {
        APPLIED, SHORTLISTED, SELECTED, REJECTED
    }
}
